source/base/MessageHandler.cpp

```diff
@@ -61,52 +61,127 @@ void MessageHandler::handleMessageReceived(Swift::Message::ref message)
     //std::cout << "handleMessageReceived: jid: " << message->getFrom() << ", bare: " << message->getFrom().toBare().toString() << ", resource: " << message->getFrom().getResource() << std::endl;
 
     unsigned int security = 0;
+    unsigned int direction = 1; // incoming
+    Swift::Forwarded::ref forwarded = nullptr;
+    Swift::Message::ref forwardedMessage = nullptr;
+    qint64 timestamp = 0;
+    QString partyJid = QString::fromStdString(message->getFrom().toBare().toString());
```

Comment:
partyJid is now set several times. And on my MaM messages, it get reset in different locations. Finally, the message is pushed into the db as a 1to1 messages with the group member, instead of a group messages. I will have a look at it and will make a proposal for a handleMessageReceived method. Please give me some time.